components/modal_controller/modal_controller.jsx

```diff
@@ -39,6 +88,11 @@ export default class ModalController extends React.Component {
         return (
             <div>
                 {modalOutput}
+                <QuickSwitchModal
```

Comment:
Instead of using the ModalStore and having this component explicitly render the QuickSwitchModal, can you look at using the `openModal` action from `actions/views/modals` to show it ([example here](https://github.com/mattermost/mattermost-webapp/blob/master/components/dot_menu/dot_menu.jsx#L139)) since that's the preferred way to control these now.

As for the hotkey, we only want that active in the normal channel view, but this component is intended to be used elsewhere. Could you move that logic somewhere else like maybe the Sidebar since that also needs to open it when you click on the button at the bottom of the channel list?

Reply:
I move back it to channel header and use open/closeModal but still need show state in parent component to handle "toggling"